Updated README.md in setup-python (#472)
actions/setup-python: README.md

# setup-python

<p align="left">
  <a href="https://github.com/actions/setup-python"><img alt="GitHub Actions status" src="https://github.com/actions/setup-python/workflows/Main%20workflow/badge.svg"></a>
</p>

> This action provides the following functionalities for GitHub Actions users:

Reviewer: ```suggestion
This action provides the following functionality for GitHub Actions users:
```

(because "functionality" here is uncountable https://en.wiktionary.org/wiki/functionality)

> - Optionally installing and adding to PATH a version of Python that is already installed in the runner's tool cache.

Reviewer: ```suggestion
- Installing a version of Python and (by default) adding it to PATH
```

Author: Thank you! Is it worth simplifying all points of the list to make them like: 
```
- Installing a version of Python or PyPy and (by default) adding it to the PATH
- Optionally caching dependencies for pip, pipenv and poetry
- Registering problem matchers for error output
```
The following paragraph of the Basic Usage section describes how action deals with tool cache and python-versions repo, so probably there is no need to put that information into the syllabus of the action?

>The action will first check the local [tool cache](docs/advanced-usage.md#hosted-tool-cache) for a [semver](https://github.com/npm/node-semver#versions) match. If unable to find a specific version in the tool cache, the action will attempt to download a version of Python from [GitHub Releases](https://github.com/actions/python-versions/releases) and for PyPy from the official [PyPy's dist](https://downloads.python.org/pypy/).

Reviewer: I agree; I like the shorter summary
